haystack/utils/import_utils.py

```diff
@@ -55,5 +60,54 @@ def get_filename_extension_from_url(url: str) -> Tuple[str, str]:
     return file_name, archive_extension
 
 
+def fetch_archive_from_http(
+    url: str,
+    output_dir: str,
+    proxies: Optional[Dict[str, str]] = None,
+    timeout: Union[float, Tuple[float, float]] = 10.0,
+) -> bool:
+    """
+    Fetch an archive (zip or gz) from a url via http and extract content to an output directory.
+    :param url: http address
+    :param output_dir: local path
+    :param proxies: proxies details as required by requests library
+    :param timeout: How many seconds to wait for the server to send data before giving up,
+        as a float, or a :ref:`(connect timeout, read timeout) <timeouts>` tuple.
+        Defaults to 10 seconds.
+    :return: if anything got fetched
+    """
+    # verify & prepare local directory
+    path = Path(output_dir)
+    if not path.exists():
```

Comment:
You can do it like this instead and avoid the `if`

```suggestion
    path.mkdir(exist_ok=True, parents=True)
```

Reply:
I'll keep it as it is, don't want to risk breaking anything.

I just copy pasted the previous version and removed the part handling tar files.